active_plugins/enforceobjectsonetoone.py

```diff
@@ -163,16 +163,18 @@ def run(self, workspace):
         workspace.display_data.statistics = []
         
         pre_primary = workspace.object_set.get_objects(self.x_name.value)
-
         pre_primary_seg = pre_primary.segmented
 
         pre_secondary = workspace.object_set.get_objects(self.y_name.value)
-
         pre_secondary_seg = pre_secondary.segmented
 
-        primary_seg = self.enforce_unique(pre_primary_seg, pre_secondary_seg, erode_excess=True)
-
-        secondary_seg = self.enforce_unique(pre_secondary_seg, primary_seg)
+        if pre_secondary.count == 0:
+            secondary_seg = primary_seg = pre_primary_seg
```

Comment:
Sorry, I don't quite follow this - if pre_secondary has nothing, why do we set everything to pre_primary (which might have something), rather than pre_secondary, which definitely won't? I believe you if you say it works, but if you wouldn't mind adding a comment, would be appreciated just in case someone else (aka "me from 6 months from now") has the same q. Thanks!

Reply:
it's because I accidentally deleted my fix when I messed up git and then rewrote it backwards...
should be correct now.